docs/src/pages/components/typography/typography.md

```diff
@@ -98,3 +98,31 @@ A few key factors to follow for an accessible typography:
 - **Color**. Provide enough contrast between text and its background, check out the minimum recommended [WCAG 2.0 color contrast ratio](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) (4.5:1).
 - **Font size**. Use [relative units (rem)](/customization/typography/#font-size) to accommodate the user's settings.
 - **Heading hierarchy**. [Don't skip](https://www.w3.org/WAI/tutorials/page-structure/headings/) heading levels. In order to solve this problem, you need to [separate the semantics from the style](#changing-the-semantic-element).
+
+## `TypographyProps` usage with TypeScript
+
+Typography supports changing of the root node with `component` prop. To be able to use `TypographyProps` on their own (for example to type custom component) with the `component` prop, `TypographyProps` should be used with type arguments. Otherwise `component` prop will not be present in `TypographyProps`.
+For example 
+```ts
+const CustomComponent: React.FC<TypographyProps<'a', { component: 'a' }>> = 
```

Comment:
We prefer `function Component(props: Props)`. anonymous functions tend to have issues with display names. `React.FC` is problematic because of implicit children. It's usually only necessary for older typescript versions

Reply:
Changed it to `function` syntax